### PR TITLE
Remove message count http examples

### PR DESCRIPTION
**Pull request by OwenCorrigan76**
If MessageCount is set in the YAMLs, run the Producer / Consumer for the set amount. If it is not set, run the Producer / Consumer indefinetly.

### CODE REVIEW

File: java/http/java-http-consumer.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ spec:
     spec:
       containers:
       - name: java-http-consumer
-        image: quay.io/strimzi-examples/java-http-consumer:latest
+        image: docker.io/owenerz/java-http-consumer:latest
```
Comment (scholzj) [inline java/http/java-http-consumer.yaml]: This should not be here.

**Review by scholzj — COMMENTED** [review]
I guess I'm fine with a high limit. But the Vert.x HTTP examples have no limit. So I really wonder what the limit usecase here is. Does anyone have any? If not, we should just remove it.

**Comment by ppatierno** [comment]
We have different behaviours here ...
The Vert.x HTTP producer, just sends indefinitely if no message count env var is set.
I think we should follow the same logic and not setting an higher value in the YAMLs.

**Comment by OwenCorrigan76** [comment]
@scholzj Considering @ppatierno 's comment, should I fix the registry directory in the consumer YAML (as outlined above) and push again with high message count?

**Comment by scholzj** [comment]
Well, you need to fix the registry in any case. But I would follow Paolo's suggestion here.

**Comment by scholzj** [comment]
I'm a bit lost about where we got and where we plan to go. I thought after Paolo's comment the plan was to remove the count or make it optional. But that does not seem to be reflected in the code.

**Comment by ppatierno** [comment]
Agree I was expecting the same.

**Comment by OwenCorrigan76** [comment]
Ah sorry. Wasn't sure whether to still remove it from the code. Now I understand. Will fix this morning.

**Comment by scholzj** [comment]
I guess you don't have to remove the config option completely, just make it optional. At least that is how I would read Paolo's suggestion and if not set have it run forever.

**Comment by ppatierno** [comment]
> I guess you don't have to remove the config option completely, just make it optional. At least that is how I would read Paolo's suggestion and if not set have it run forever.

Yes, that is what I mean. Option to specify should stay. I would remove the configuration of the env var in the YAML and leaving by default to send messages indefinitely.

File: java/http/java-http-producer/src/main/java/HttpProducer.java [hunk 2]
```diff
@@ -64,17 +64,27 @@ public HttpProducer(HttpProducerConfig config) throws URISyntaxException {
                 GlobalOpenTelemetry.getTracer("client-examples");
     }
 
+    private Long getDefaultMessageCount() {
+        Long messageCount = this.config.getMessageCount();
+        return messageCount == null ? 1000000L : messageCount;
+    }
```
Comment (scholzj) [inline java/http/java-http-producer/src/main/java/HttpProducer.java]: Same as above -> why not just run forever?

File: java/http/java-http-consumer/src/main/java/HttpConsumer.java [hunk 3]
```diff
@@ -121,19 +121,27 @@ public void subscribe() throws URISyntaxException, IOException, InterruptedExcep
     }
 
     public void run() throws InterruptedException {
+        Long messageCount = this.config.getMessageCount();
+        if ( messageCount == null ) {
+            messageCount = (Long.MAX_VALUE / this.config.getPollInterval() - 60_000L);
+        }
```
Comment (scholzj) [inline java/http/java-http-consumer/src/main/java/HttpConsumer.java]: What not just run forever when it is null?

**Review by scholzj — COMMENTED** [review]
TBH, I'm a bit confused where we are heading with this one based on the description and on the discussion.

File: java/http/java-http-producer/src/main/java/HttpProducer.java (hunk 2 above)
Comment (OwenCorrigan76) [reply java/http/java-http-producer/src/main/java/HttpProducer.java:70]: Honestly, I don't know how. I'll look at it again.

Comment (OwenCorrigan76) [reply java/http/java-http-producer/src/main/java/HttpProducer.java:70]: Would a while loop be appropriate here?

Comment (OwenCorrigan76) [reply java/http/java-http-producer/src/main/java/HttpProducer.java]: Above now removed

File: java/http/java-http-consumer/src/main/java/HttpConsumer.java (hunk 3 above)
Comment (OwenCorrigan76) [reply java/http/java-http-consumer/src/main/java/HttpConsumer.java]: Above now removed

**Comment by OwenCorrigan76** [comment]
Sorry this took so long. Now, using CountDownLatch, the Producer / Consumer runs for desired amount when STRIMZI_MESSAGE_COUNT is supplied, and runs forever if not supplied.

File: java/http/deployment-tracing-opentelemetry.yaml [hunk 4]
```diff
@@ -26,6 +26,8 @@ spec:
             value: my-topic
           - name: STRIMZI_SEND_INTERVAL
             value: "1000"
+          - name: STRIMZI_MESSAGE_COUNT
+            value: "1000000"
```
Comment (ppatierno) [inline java/http/deployment-tracing-opentelemetry.yaml]: Didn't we say the default behaviour will be sending forever? So why putting this so high value if now not setting the env var means sending forever?

File: java/http/deployment-tracing-opentelemetry.yaml [hunk 5]
```diff
@@ -70,6 +72,8 @@ spec:
             value: "my-group"
           - name: STRIMZI_POLL_INTERVAL
             value: "1000"
+          - name: STRIMZI_MESSAGE_COUNT
+            value: "1000000"
```
Comment (ppatierno) [inline java/http/deployment-tracing-opentelemetry.yaml]: Ditto as above but for consuming indefinitely?

File: java/http/deployment-tracing-opentelemetry.yaml (hunk 4 above)
Comment (OwenCorrigan76) [reply java/http/deployment-tracing-opentelemetry.yaml]: Ah, sorry. I missed this one. Fixing now.

File: java/http/deployment-tracing-opentelemetry.yaml (hunk 5 above)
Comment (OwenCorrigan76) [reply java/http/deployment-tracing-opentelemetry.yaml]: Removed. I forgot about the otel YAMLs. Tested without STRIMZI_MESSAGE_COUNT and working as expected.

**Comment by OwenCorrigan76** [comment]
OpenTelemetry YAMLs updated.

File: java/http/java-http-consumer/src/main/java/HttpConsumer.java [hunk 6]
```diff
@@ -123,17 +129,20 @@ public void subscribe() throws URISyntaxException, IOException, InterruptedExcep
     public void run() throws InterruptedException {
         log.info("Scheduling periodic poll every {} ms waiting for {} ...", this.config.getPollInterval(), this.config.getMessageCount());
         this.executorService.scheduleAtFixedRate(this::scheduledPoll, 0, this.config.getPollInterval(), TimeUnit.MILLISECONDS);
-        this.executorService.awaitTermination(this.config.getPollInterval() * this.config.getMessageCount() + 60_000L, TimeUnit.MILLISECONDS);
         log.info("... {} messages received", this.messageReceived);
     }
 
+
```
Comment (ppatierno) [inline java/http/java-http-consumer/src/main/java/HttpConsumer.java]: why this extra line?

File: java/http/java-http-consumer/src/main/java/HttpConsumer.java [hunk 7]
```diff
@@ -123,17 +129,20 @@ public void subscribe() throws URISyntaxException, IOException, InterruptedExcep
...
         this.poll();
-        if (this.messageReceived == this.config.getMessageCount()) {
+        if (this.config.getMessageCount() != null && this.messageReceived == this.config.getMessageCount()) {
             this.executorService.shutdown();
+            this.messagesReceivedLatch.countDown();
+            log.info("All messages received");
         }
     }
 
+
```
Comment (ppatierno) [inline java/http/java-http-consumer/src/main/java/HttpConsumer.java]: why this extra line?

File: java/http/java-http-consumer/src/main/java/HttpConsumer.java [hunk 8]
```diff
@@ -167,11 +176,14 @@ public void poll() {
                 span.setAttribute(SemanticAttributes.HTTP_STATUS_CODE, response.statusCode());
                 span.setStatus(response.statusCode() == 200 ? StatusCode.OK : StatusCode.ERROR);
             } finally {
+
```
Comment (ppatierno) [inline java/http/java-http-consumer/src/main/java/HttpConsumer.java]: why this extra line?

File: java/http/java-http-consumer/src/main/java/HttpConsumer.java [hunk 9]
```diff
@@ -167,11 +176,14 @@ public void poll() {
...
                 span.end();
             }
         } catch (URISyntaxException | IOException | InterruptedException e) {
             throw new RuntimeException(e);
         }
+
+
```
Comment (ppatierno) [inline java/http/java-http-consumer/src/main/java/HttpConsumer.java]: why this extra lines?

File: java/http/java-http-consumer/src/main/java/HttpConsumer.java [hunk 10]
```diff
@@ -189,3 +201,4 @@ public void deleteConsumer() throws IOException, InterruptedException {
         }
     }
 }
+
```
Comment (ppatierno) [inline java/http/java-http-consumer/src/main/java/HttpConsumer.java]: why this extra line?

**Review by ppatierno — CHANGES_REQUESTED** [review]

**Comment by OwenCorrigan76** [comment]
@ppatierno Apologies. Extra lines were just a sloppy oversight. Removed now.

**Review by ppatierno — APPROVED** [review]
LGTM

**Review by scholzj — APPROVED** [review]

**Pull request merged**